File: reorganize_modal.js
const fs = require('fs');
const path = require('path');

const filePath = process.argv[2];
let content = fs.readFileSync(filePath, 'utf8');

// Define the sections to extract
// We'll use marker comments or specific HTML blocks as delimiters

const datesStartMarker = '<!-- Order: Cancellation, Elapse, Contract -->';
const datesEndMarker = '</div>\r\n\r\n            <!-- Title & Status Section -->';
// Wait, the markers might not match exactly. Let me use more robust regex.

const getSection = (startRegex, endRegex) => {
    const startMatch = content.match(startRegex);
    if (!startMatch) return null;
    const startIndex = startMatch.index;
    const rest = content.substring(startIndex + startMatch[0].length);
    const endMatch = rest.match(endRegex);
    if (!endMatch) return null;
    const endIndex = startIndex + startMatch[0].length + endMatch.index + endMatch[0].length;
    return {
        full: content.substring(startIndex, endIndex),
        start: startIndex,
        end: endIndex
    };
};

// 1. Get Date Header
const datesSection = getSection(/<!-- Order: Cancellation, Elapse, Contract -->/, /<\/div>\s*<\/div>/);
if (!datesSection) { console.error("Could not find Dates Section"); process.exit(1); }

// 2. Get Name/Status
const nameSection = getSection(/<!-- Title & Status Section -->/, /<\/div>\s*<\/div>/);
if (!nameSection) { console.error("Could not find Name Section"); process.exit(1); }

// 3. Get Financial Board
const financialSection = getSection(/\${hasFinancials \? `/, /` : ''}/);
if (!financialSection) { console.error("Could not find Financial Section"); process.exit(1); }

// 4. Get Info Sections
const infoSection = getSection(/<!-- Information Sections -->/, /<\/div>\s*<\/div>\s*`;/);
if (!infoSection) { console.error("Could not find Info Section"); process.exit(1); }

// Reconstruct the template body
// New Order: Name -> Financial -> Dates -> Info

const templateStart = content.indexOf('<div class="p-6 sm:p-8">');
const templateEnd = content.indexOf('</div>\r\n    `;', templateStart) + 8; // Adjust based on file

// Actually, let's just replace the blocks in-place but that's messy.
// Let's replace the whole body of <div class="p-6 sm:p-8"> ... </div>

const bodyStart = templateStart + '<div class="p-6 sm:p-8">'.length;
const bodyEnd = templateEnd - '</div>\r\n    `;'.length;

const newBody = `
            <!-- Section 1: Title, Status & Financial Board -->
            <div class="mb-10">
                ${nameSection.full}
                ${financialSection.full}
            </div>

            <!-- Section 2: Integrated Key Dates -->
            <div class="mb-10 pb-8 border-b border-slate-100">
                ${datesSection.full}
            </div>

            <!-- Section 3: Information Sections -->
            ${infoSection.full}
        `;

const finalContent = content.substring(0, bodyStart) + newBody + content.substring(bodyEnd);

fs.writeFileSync(filePath, finalContent);
console.log("Successfully reorganized app.js");

File: reorganize_modal.py
import os
import re
import sys

file_path = sys.argv[1]
with open(file_path, 'r', encoding='utf-8') as f:
    content = f.read()

# 1. Get Date Header Section
# Matches from <!-- Order: ... --> to </div>
dates_pattern = r'(            <!-- Order: Cancellation, Elapse, Contract -->\s*<div class="flex flex-wrap gap-3 mb-8 pb-8 border-b border-slate-100">.*?</div>\s*' + re.escape('    `') + r'\s*: \'\')\s*</div>'
# That's too complex. Let's use simple find/replace with known strings.

dates_start = '<!-- Order: Cancellation, Elapse, Contract -->'
dates_end = '</div>\n            </div>' # This is tricky due to nested divs

# Let's use a simpler approach: get indices based on UNIQUE markers
i_dates_start = content.find('<!-- Order: Cancellation, Elapse, Contract -->')
i_dates_div_start = content.find('<div class="flex flex-wrap gap-3 mb-8 pb-8 border-b border-slate-100">', i_dates_start)
# Find the matching </div> for i_dates_div_start
def find_matching_div(s, start_idx):
    depth = 0
    i = start_idx
    while i < len(s):
        if s[i:i+4] == '<div':
            depth += 1
            i += 4
        elif s[i:i+6] == '</div>':
            depth -= 1
            i += 6
            if depth == 0:
                return i
        else:
            i += 1
    return -1

i_dates_end = find_matching_div(content, i_dates_div_start)
dates_block = content[i_dates_start:i_dates_end]

# 2. Get Name/Status Section
i_name_start = content.find('<!-- Title & Status Section -->')
i_name_div_start = content.find('<div class="mb-8">', i_name_start)
i_name_end = find_matching_div(content, i_name_div_start)
name_block = content[i_name_start:i_name_end]

# 3. Get Financial Section
i_fin_start = content.find('${hasFinancials ? `')
i_fin_end = content.find('` : \'\'}', i_fin_start) + 7
fin_block = content[i_fin_start:i_fin_end]

# 4. Get Info Sections
i_info_start = content.find('<!-- Information Sections -->')
i_info_div_start = content.find('<div class="space-y-2">', i_info_start)
i_info_end = find_matching_div(content, i_info_div_start)
info_block = content[i_info_start:i_info_end]

# Reconstruct
# We want: Name -> Financial -> Dates -> Info
# The container is <div class="p-6 sm:p-8">

container_start = content.find('<div class="p-6 sm:p-8">') + len('<div class="p-6 sm:p-8">')
container_end = content.rfind('</div>', 0, content.find('`;', container_start))

new_body = f"""
            <!-- Section 1: Title, Status & Financial Board -->
            <div class="mb-10">
                {name_block}
                {fin_block}
            </div>

            <!-- Section 2: Integrated Key Dates -->
            <div class="mb-10 pb-8 border-b border-slate-100">
                {dates_block}
            </div>

            <!-- Section 3: Information Sections -->
            <div class="space-y-4">
                ${{renderSection(sections.profile)}}
                ${{renderSection(sections.admin)}}
                ${{renderSection(sections.property)}}
            </div>
"""

final_content = content[:container_start] + new_body + content[container_end:]

with open(file_path, 'w', encoding='utf-8') as f:
    f.write(final_content)

print("Successfully reorganized app.js with Python")
